src/hotspot/share/opto/superword.cpp

```diff
@@ -4095,14 +3870,35 @@ SWPointer::SWPointer(MemNode* mem, SuperWord* slp, Node_Stack *nstack, bool anal
   _debug_invar(nullptr), _debug_negate_invar(false), _debug_invar_scale(nullptr),
 #endif
   _nstack(nstack), _analyze_only(analyze_only),
-  _stack_idx(0)
+  _stack_idx(0), _phase(slp->phase()), _lpt(slp->lpt())
 #ifndef PRODUCT
   , _tracer(slp)
 #endif
 {
-  NOT_PRODUCT(_tracer.ctor_1(mem);)
+  init();
+}
 
-  Node* adr = mem->in(MemNode::Address);
+// Following is used outside superword optimization
```

Comment:
Could we move the whole SWPointer outside of SuperWord, into some "autovectorization.hpp" maybe? Because the SW of SWPointer means SuperWord, maybe a renaming could be good too?

Reply:
If you are going to do that, I'd suggest doing this refactoring in a separate RFE. It would help in general with any future extension to auto-vectorization.

Reply:
Can we untangle it completely from SuperWord? it seems you have made it optional, so yes. And maybe we can also make the trace flags like `_slp->is_trace_alignment()` independent? It would be nice to also be able to trace this for non SuperWord-contexts like post-loop masked vectoriaztion, right?

Reply:
I will try to do this in another JBS and come back here later.

Reply:
That would be fantastic!

Reply:
`SWPointer` is already moved out from `SuperWord` in [JDK-8312332](https://bugs.openjdk.org/browse/JDK-8312332). We will move `_vector_loop_debug` out in a later refactoring patch.